experimental/Schemes/BlowupMorphism.jl

```diff
@@ -96,108 +96,236 @@ function strict_transform(p::BlowupMorphism, inc::CoveredClosedEmbedding)
+#
+#For a 'BlowupMorphism'  ``p : Y → X`` and an IdealSheaf ''I'' on ''X'' return the
+#strict transform of ''I'' on ''Y''.
+#"""
 function strict_transform(p::BlowupMorphism, I::IdealSheaf)
+  Istrict,_ =_do_transform(p,I,-1)
+  return Istrict
+end
+
+#@Markdown.doc """
+#  weak_transform(p::BlowupMorphism, I::IdealSheaf)
+#
+#For a 'BlowupMorphism'  ``p : Y → X`` and an IdealSheaf ''I'' on ''X'' return the
+#weak transform of ''I'' on ''Y'' and the multiplicity of the exceptional divisor in the total transform of ''I''.
+#"""
+function weak_transform(p::BlowupMorphism, I::IdealSheaf)
+  Iweak,_ =_do_transform(p,I,0)
+  return Iweak
+end
+
+function weak_transform_decorated(p::BlowupMorphism, I::IdealSheaf)
+  Iweak, multi = _do_transform(p,I,0)
+  return Iweak,multi
+end
+
+#@Markdown.doc """
+#  controlled_transform(p::BlowupMorphism, I::IdealSheaf, b::int)
+#
+#For a 'BlowupMorphism'  ``p : Y → X`` and an IdealSheaf ''I'' on ''X'' return the
+#controlled transform of ''I'' on ''Y'' with control ''b''.
+#"""
+function controlled_transform(p::BlowupMorphism, I::IdealSheaf, b::Int)
+  Icontrol,_ = _do_transform(p,I,b)
+  return Icontrol
+end
+
+##########################################################################################################
+## central internal method for strict, weak and controlled transforms of IdealSheafs and subschemes
+##########################################################################################################
+function _do_transform(p::BlowupMorphism, I::IdealSheaf, method::Int=-1)
+## method: -1  strict transform
+##          0  weak transform
+##         b>0  controlled transform with control b>0
+##         < -1 error
+  method > -2  || error("unknown method of transform", method)
+
+  ## initializations and sanity checks for p
   X = scheme(I)
   Y = domain(p) 
   X === codomain(p) || error("ideal sheaf is not defined on the codomain of the morphism")
-
+  IE = ideal_sheaf(exceptional_divisor(p))
   ID = IdDict{AbsSpec, Ideal}()
+
+  ## get our hands on the coverings -- using simplified covering for CY
   pr = projection(p)
-  p_cov = covering_morphism(pr)
-  CY = domain(p_cov)
-  # We first apply elim_part to all the charts.
-# CY_simp, phi, psi = simplify(CY)
-# # register the simplification in Y
-# push!(coverings(Y), CY_simp)
-# refinements(Y)[(CY_simp, CY)] = phi
-# refinements(Y)[(CY, CY_simp)] = psi
-# CY === default_covering(Y) && set_attribute!(Y, :simplified_covering, CY_simp)
-  CY_simp = (CY === default_covering(Y) ? simplified_covering(Y) : CY)
-  phi = (CY === default_covering(Y) ? Y[CY_simp, CY] : identity_map(CY_simp))
-
-  # compose the covering morphisms
-  p_cov_simp = compose(phi, p_cov)
-  CX = codomain(p_cov)
-  E = exceptional_divisor(p)
+  p_cov_temp = covering_morphism(pr)
+  CX = codomain(p_cov_temp)
+  CY = domain(p_cov_temp)
+## CAUTION:
+## simplification already taken into account, but switched off in has_simplified_covering!!!!
```

Comment:
Is this comment deprecated by now?

Reply:
I am still fighting with simplification. If you could help with this sometime during this week, I would be greatful.

Reply:
Now this comment is deprecated and has been removed.